src/main/java/com/theelixrlabs/healthcare/service/DoctorService.java

```diff
@@ -2,41 +2,63 @@
 
 import com.theelixrlabs.healthcare.constants.DoctorConstants;
 import com.theelixrlabs.healthcare.constants.MessageConstants;
+import com.theelixrlabs.healthcare.dao.DoctorPatientAssignmentRepository;
 import com.theelixrlabs.healthcare.exceptionHandler.CustomException;
 import com.theelixrlabs.healthcare.exceptionHandler.ResourceNotFoundException;
 import com.theelixrlabs.healthcare.model.DoctorModel;
+import com.theelixrlabs.healthcare.model.DoctorPatientAssignmentModel;
 import com.theelixrlabs.healthcare.repository.DoctorRepository;
 import com.theelixrlabs.healthcare.dto.DoctorDto;
 import com.theelixrlabs.healthcare.validation.DoctorModelValidator;
 import com.theelixrlabs.healthcare.utility.MessageUtil;
 import com.theelixrlabs.healthcare.validation.Validator;
 import org.springframework.stereotype.Service;
+
+import java.util.List;
 import java.util.Optional;
 import java.util.ArrayList;
-import java.util.List;
 import java.util.UUID;
 
 /**
  * Service class responsible for handling operations related to doctor.
  */
 @Service
 public class DoctorService {
+
     private final DoctorRepository doctorRepository;
+
     private final DoctorModelValidator doctorModelValidator;
+
+    private final DoctorPatientAssignmentRepository doctorPatientAssignmentRepository;
+
     private final MessageUtil messageUtil;
+
     private final Validator validator;
 
-    public DoctorService(DoctorRepository doctorRepository, MessageUtil messageUtil, Validator validator,DoctorModelValidator doctorModelValidator) {
+    //Constructor Injection.
+    public DoctorService(DoctorRepository doctorRepository, DoctorPatientAssignmentRepository doctorPatientAssignmentRepository, DoctorModelValidator doctorModelValidator, MessageUtil messageUtil, Validator validator) {
         this.doctorRepository = doctorRepository;
+        this.doctorPatientAssignmentRepository = doctorPatientAssignmentRepository;
         this.doctorModelValidator = doctorModelValidator;
         this.messageUtil = messageUtil;
         this.validator = validator;
     }
 
+    /**
+     * Checks if a doctor is assigned to any patients.
+     *
+     * @param validDoctorId    The unique identifier of the doctor to check.
+     * @return true if the doctor is assigned to at least one patient, false otherwise.
+     */
+    private boolean isDoctorAssignedToPatient(UUID validDoctorId) {
+        List<DoctorPatientAssignmentModel> doctorAssignmentList = doctorPatientAssignmentRepository.findByDoctorId(validDoctorId);
```

Comment:
instead retrieving list why can't you just use existBy
Also this list may contain assigned and unassigned data as-well 
Please make use of a correct repo method

Reply:
Resolved.